lib/rucio/daemons/bb8/common.py

```diff
@@ -171,17 +245,24 @@ def select_target_rse(current_rse, rse_expression, subscription_id, rse_attribut
     #     pass
     #     # get_subscription_by_id(subscription_id, session)
     if force_expression is not None:
-        rses = parse_expression(expression='(%s)\\%s' % (force_expression, target_rse), filter={'availability_write': True}, session=session)
+        if parent_rule['grouping'] != RuleGrouping.NONE:
+            rses = parse_expression(expression='(%s)\\%s' % (force_expression, target_rse), filter={'availability_write': True}, session=session)
+        else:
+            # in order to avoid replication of the part of distributed dataset not present at rabalanced rse -> rses in force_expression
+            # this will be extended with development of delayed rule
+            rses = parse_expression(expression='((%s)|(%s))\\%s' % (force_expression, rse_expression, target_rse), filter={'availability_write': True}, session=session)
     elif len(rses) > 1:
         # Just define the RSE Expression without the current_rse
         return '(%s)\\%s' % (rse_expression, target_rse)
-    elif rse_attributes['tier'] is True or rse_attributes['tier'] == '1':
+    elif rse_attributes['tier'] is True or int(rse_attributes['tier']) == 1:
```

Comment:
what happens if an RSE does not have the attribute 'tier' set?

Reply:
I get a KeyError, but it used to be there since ever. I'll fix that.

Reply:
This is not easy. We have no such a case for our DATADISKs. If other customers won't use our notation of DATADISKs and tiers, this all will be completely broken and needed to be completely rewritten. There might be three solutions for now, 1) leave it be, 2) if tier not in attributes: rebalance to all datadisks 3) if tier not in attributes: skip the rse. I would prefer the first one for the moment.

Reply:
ok, but please put a TODO comment in the code so we don't forget